satpy/readers/vii_l1b_nc.py

```diff
@@ -69,19 +71,18 @@ def _perform_calibration(self, variable, dataset_info):
         if calibration_name == 'brightness_temperature':
             # Extract the values of calibration coefficients for the current channel
             chan_index = dataset_info['chan_thermal_index']
-            cw = self._channel_cw_thermal[chan_index] * 1e-3
+            cw = self._channel_cw_thermal[chan_index]  # wavelengths now in µm
             a = self._bt_conversion_a[chan_index]
             b = self._bt_conversion_b[chan_index]
             # Perform the calibration
             calibrated_variable = self._calibrate_bt(variable, cw, a, b)
             calibrated_variable.attrs = variable.attrs
         elif calibration_name == 'reflectance':
-            scale = 1/(dataset_info['wavelength'][2] - dataset_info['wavelength'][0])
             # Extract the values of calibration coefficients for the current channel
             chan_index = dataset_info['chan_solar_index']
-            isi = scale * self._integrated_solar_irradiance[chan_index]
+            isi = self._integrated_solar_irradiance[chan_index]  # Band averaged solar irradiance in (W/m2/µm)
             # Perform the calibration
-            calibrated_variable = self._calibrate_refl(variable, self.angle_factor, isi)
+            calibrated_variable = self._calibrate_refl(variable, self.angle_factor.values, isi)
```

Comment:
the `.values` is triggering a computation and breaking the laziness of the calibration process. Is this really necessary?

Reply:
I couldn't get it to work without the .values. I am not really sure why. I will check again.

Reply:
It looks like the calibrate_refl function does not like the coords part of the angle_factor. If I pass angle_factor.values, it works fine, or I can pass angle_factor and just use angle_factor.values within the reflectance calculation itself (i.e. something like refl_values = (np.pi / isi) * angle_factor.values * radiance * 100.0). Which is better?

Reply:
Does it work if you use `.data` instead of `.values`?

Reply:
Yes it does, shall I use that instead?